libpromises/promises.c

```diff
@@ -424,9 +425,43 @@ Promise *DeRefCopyPromise(EvalContext *ctx, const Promise *pp)
         SeqDestroy(bodies_and_args);
     }
 
+    // Add default body for promise body types that are not present
+    char *bundle_type = pcopy->parent_promise_type->parent_bundle->type;
+    char *promise_type = pcopy->parent_promise_type->name;
+    const PromiseTypeSyntax *syntax = PromiseTypeSyntaxGet(bundle_type, promise_type);
+    AddDefaultBodiesToPromise(ctx, pcopy, syntax);
+
+    // Add default body for global body types that are not present
+    const PromiseTypeSyntax *global_syntax = PromiseTypeSyntaxGet("*", "*");
+    AddDefaultBodiesToPromise(ctx, pcopy, global_syntax);
+
     return pcopy;
 }
 
+// Try to add default bodies to promise for every body type found in syntax
+static void AddDefaultBodiesToPromise(EvalContext *ctx, Promise *promise, const PromiseTypeSyntax *syntax)
+{
+    // iterate over possible constraints
+    for (int i = 0; syntax->constraints[i].lval; i++)
+    {
+        // of type body
+        if(syntax->constraints[i].dtype == CF_DATA_TYPE_BODY) {
+            const char *constraint_type = syntax->constraints[i].lval;
+            // if there is no matching body in this promise
+            if(!PromiseBundleOrBodyConstraintExists(ctx, constraint_type, promise)) {
+                const Policy *policy = PolicyFromPromise(promise);
+                // default format is <promise_type>_<body_type>_default
+                char* default_body_name = StringConcatenate(4, "default:", promise->parent_promise_type->name, "_", constraint_type);
+                const Body *bp = EvalContextFindFirstMatchingBody(policy, constraint_type, "default", default_body_name);
+                if(bp) {
+                  printf("Found %s\n", default_body_name);
```

Comment:
You should log in `VERBOSE` mode so users know what's happening.  @ncharles had concerns about this implicit behavior so we need to tell the user that something happened here.